tests/utils/lambda-helpers.mjs

```diff
@@ -0,0 +1,197 @@
+ * @property {Record<string, unknown>} [flags] Feature flags that should be set during the invocation
+ *
+ * @typedef {Pick<FunctionInvocationOptions, 'env'>} LoadFunctionOptions
+ */
+
+/**
+ * This is helper to get LambdaLocal's execute to actually provide result type instead of `unknown`
+ * Because jsdoc doesn't seem to have equivalent of `as` in TS and trying to assign `LambdaResult` type
+ * to return value of `execute` leading to `Type 'unknown' is not assignable to type 'LambdaResult'`
+ * error, this types it as `any` instead which allow to later type it as `LambdaResult`.
+ * @param  {Parameters<typeof untypedExecute>} args
+ * @returns {Promise<LambdaResult>}
+ */
+async function execute(...args) {
+  /**
+   * @type {any}
+   */
+  const anyResult = await untypedExecute(...args)
+
+  return anyResult
+}
+
+/**
+ * @param {FixtureTestContext} ctx
+ */
+export const createBlobContext = (ctx) =>
+  Buffer.from(
+    JSON.stringify({
+      edgeURL: `http://${ctx.blobStoreHost}`,
+      uncachedEdgeURL: `http://${ctx.blobStoreHost}`,
+      token: BLOB_TOKEN,
+      siteID: ctx.siteID,
+      deployID: ctx.deployID,
+      primaryRegion: 'us-test-1',
+    }),
+  ).toString('base64')
+
+/**
+ * Converts a readable stream to a buffer
+ * @param {NodeJS.ReadableStream} stream
+ * @returns {Promise<Buffer>}
+ */
+function streamToBuffer(stream) {
+  /**
+   * @type {Buffer[]}
+   */
+  const chunks = []
+
+  return new Promise((resolve, reject) => {
+    stream.on('data', (chunk) => chunks.push(Buffer.from(chunk)))
+    stream.on('error', (err) => reject(err))
+    stream.on('end', () => resolve(Buffer.concat(chunks)))
+  })
+}
+
+/**
+ * @param {FixtureTestContext} ctx
+ * @param {Record<string, string>} [env]
+ */
+function temporarilySetEnv(ctx, env) {
+  const environment = {
+    NODE_ENV: 'production',
+    NETLIFY_BLOBS_CONTEXT: createBlobContext(ctx),
+    ...(env || {}),
+  }
+
+  const envVarsToRestore = {}
+
+  // We are not using lambda-local's environment variable setting because it cleans up
+  // environment vars to early (before stream is closed)
+  Object.keys(environment).forEach(function (key) {
+    if (typeof process.env[key] !== 'undefined') {
+      envVarsToRestore[key] = process.env[key]
+    }
+    process.env[key] = environment[key]
+  })
+
+  return function restoreEnvironment() {
+    Object.keys(environment).forEach(function (key) {
+      if (typeof envVarsToRestore[key] !== 'undefined') {
+        process.env[key] = envVarsToRestore[key]
+      } else {
+        delete process.env[key]
+      }
+    })
+  }
+}
+
+const DEFAULT_FLAGS = {}
+
+/**
+ * @param {FixtureTestContext} ctx
+ * @param {LoadFunctionOptions} options
+ */
+export async function loadFunction(ctx, { env } = {}) {
+  const restoreEnvironment = temporarilySetEnv(ctx, env)
+
+  const { handler } = await import(
+    'file:///' + join(ctx.functionDist, SERVER_HANDLER_NAME, '___netlify-entry-point.mjs')
+  )
```

Comment:
note that this changed order versus current implementation so that env vars are set BEFORE importing serverless handler

this is because some of the imported code evaluate env vars at import time - so it's pretty helpful to set them up early (for example when using `NEXT_PRIVATE_DEBUG_CACHE`)